Application/Security.js

```diff
@@ -100,18 +100,117 @@ angular.module('shared')
           }
         );
       },
+      oAuth2Authorize: function(client_id, redirect_uri, response_type, scope, user, callback) {
+        var self = this;
+        
+        switch (response_type) {
```

Comment:
I've updated the PR to include stubs and validation for other oAuth2 grant types. As far as the more promise-oriented workflow, you're right, I didn't realize that $http returns promises. I'll rework this. Thanks!